src/query/boolean_query/boolean_weight.rs

```diff
@@ -163,15 +210,19 @@ impl<TScoreCombiner: ScoreCombiner> BooleanWeight<TScoreCombiner> {
 impl<TScoreCombiner: ScoreCombiner + Sync> Weight for BooleanWeight<TScoreCombiner> {
     fn scorer(&self, reader: &SegmentReader, boost: Score) -> crate::Result<Box<dyn Scorer>> {
         if self.weights.is_empty() {
-            Ok(Box::new(EmptyScorer))
-        } else if self.weights.len() == 1 {
+            return Ok(Box::new(EmptyScorer));
+        }
+        if self.minimum_number_should_match > self.weights.len() {
+            return Ok(Box::new(EmptyScorer));
```

Comment:
that's correct, but we can be more accurate by looking at the number of should sub query